nixos/modules/services/monitoring/prometheus/exporters/restic.nix

```
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.restic;
+in
+{
+  port = 9753;
+  extraOpts = {
+    repository = mkOption {
+      type = types.str;
+      description = lib.mdDoc ''
+        repository to monitor.
+      '';
+      example = "sftp:backup@192.168.1.100:/backups/example";
+    };
+
+    passwordFile = mkOption {
+      type = types.str;
+      description = lib.mdDoc ''
+        Read the repository password from a file.
+      '';
+      example = "/etc/nixos/restic-password";
+    };
+
+    environmentFile = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = lib.mdDoc ''
+        file containing the credentials to access the repository, in the
+        format of an EnvironmentFile as described by systemd.exec(5)
+      '';
+    };
+
+    rcloneOptions = mkOption {
+      type = with types; nullOr (attrsOf (oneOf [ str bool ]));
+      default = null;
+      description = lib.mdDoc ''
+        Options to pass to rclone to control its behavior.
+        See <https://rclone.org/docs/#options> for
+        available options. When specifying option names, strip the
+        leading `--`. To set a flag such as
+        `--drive-use-trash`, which does not take a value,
+        set the value to the Boolean `true`.
+      '';
+    };
+
+    rcloneConfig = mkOption {
+      type = with types; nullOr (attrsOf (oneOf [ str bool ]));
+      default = null;
+      description = lib.mdDoc ''
+        Configuration for the rclone remote being used for backup.
+        See the remote's specific options under rclone's docs at
+        <https://rclone.org/docs/>. When specifying
+        option names, use the "config" name specified in the docs.
+        For example, to set `--b2-hard-delete` for a B2
+        remote, use `hard_delete = true` in the
+        attribute set.
+        Warning: Secrets set in here will be world-readable in the Nix
+        store! Consider using the `rcloneConfigFile`
+        option instead to specify secret values separately. Note that
+        options set here will override those set in the config file.
```

Have a look at https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#syntax

The {type}\`var\` syntax can be used a lot of other places in the docstrings as well.

```suggestion

        ::: {.warning}
        Secrets set in here will be world-readable in the Nix
        store! Consider using the {option}`rcloneConfigFile`
        option instead to specify secret values separately. Note that
        options set here will override those set in the config file.
        :::
```